### PR TITLE
Use RequireJS Optimizer to minify JS and CSS

### PR DESCRIPTION
**Pull request by rlucioni**
Yields minified files with unhashed names we can use to fetch these assets from Studio. ECOM-2862.

@AlasdairSwan please review. @jimabramson FYI.

### CODE REVIEW

File: build.js [hunk 1]
```diff
@@ -1,20 +1,44 @@
...
     wrapShim: true,
 
-    // Disable all optimization. django-compressor will handle that for us.
-    optimizeCss: false,
-    optimize: 'none',
+    // See: https://github.com/jrburke/r.js/blob/master/build/example.build.js#L208-L220
+    optimizeCss: 'standard',
+
+    // See: https://github.com/jrburke/r.js/blob/master/build/example.build.js#L99-L110
+    optimize: 'uglify2',
```
Comment (feanil) [inline build.js]: How does uglify get installed?

**Comment by feanil** [comment]
If your assets are un-hashed this will prevent future use of CDNs with programs.  Is there a different part of the path that is going to get hashed or is this a known tradeoff?

File: build.js (hunk 1 above)
Comment (rlucioni) [inline build.js]: @feanil I believe it's included with r.js. @AlasdairSwan correct me if I'm mistaken.

**Comment by rlucioni** [comment]
@feanil Django Compressor is still being run, giving us hashed assets to work with if and when we need them. In the meantime, the use of unhashed assets doesn't preclude the use of cache-control headers. I'll look into extending https://github.com/edx/configuration/pull/2602 to add these headers to Programs' Nginx configuration.

**Comment by AlasdairSwan** [comment]
:+1:

**Pull request merged**